src/muc_light/mod_muc_light_utils.erl

```diff
@@ -237,90 +231,123 @@ value2b(Val, float) -> float_to_binary(Val).
 
 %% ---------------- Affiliations manipulation ----------------
 
--spec fix_edge_aff_users_case(ChangeResult :: change_aff_success() | {error, bad_request}) ->
+-spec maybe_select_new_owner(ChangeResult :: change_aff_success() | {error, bad_request}) ->
+    change_aff_success() | {error, bad_request}.
+maybe_select_new_owner({ok, AU, AUC, JoiningUsers, LeavingUsers} = AffRes) ->
+    {AffUsers, AffUsersChanged} =
+        case is_new_owner_needed(AU) of
+            true ->
+                {NewOwner, PromotionType} = find_new_owner(AU, AUC, JoiningUsers),
+                NewAU = lists:keyreplace(NewOwner, 1, AU, {NewOwner, owner}),
+                NewAUC = update_au(PromotionType, NewOwner, AUC),
+                {NewAU, NewAUC};
+            false ->
+                {AU, AUC}
+        end,
+    {ok, AffUsers, AffUsersChanged, JoiningUsers, LeavingUsers};
+maybe_select_new_owner(Error) ->
+    Error.
+
+update_au(promote_old_member, NewOwner, AUC) ->
```

Comment:
your original name suggestion for this function is 100% correct - `update_auc`.  please change it back.
AU is Affiliated Users, and AUC is - Affiliated Users Changes. 
AUC contains changes that must be made to the `original AU` (before the changes) to get newly `generated AU` (after the changes).
this exact function doesn't change AU (it's done at line 241), but it updates AUC to reflect AU change.